src/webui/webapplication.cpp

```diff
@@ -512,8 +512,8 @@ QString WebApplication::generateSid() const
         const size_t size = 6;
         quint32 tmp[size];
 
-        for (size_t i = 0; i < size; ++i)
-            tmp[i] = Utils::Random::rand();
+        for (auto &entry : tmp)
+            entry = Utils::Random::rand();
 
         sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(quint32) * size).toBase64();
```

Comment:
Or rewrite this block to:
```c++
const quint32 tmp[] = {Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand(), Utils::Random::rand()};
sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(tmp)).toBase64();
```

Reply:
Would `using namespace Utils;` be acceptable here to reduce the length?

Reply:
>Would `using namespace Utils;` be acceptable here to reduce the length?

should be ok, however I don't really think there is such need, seldom we need to read that line carefully.
here is a even shorter version:
```c++
const auto r = []() { return Utils::Random::rand(); };
const quint32 tmp[] = {r(), r(), r(), r(), r(), r()};
sid = QByteArray::fromRawData(reinterpret_cast<const char *>(tmp), sizeof(tmp)).toBase64();
```

Reply:
>here is a even shorter version:

Do you really think it's a good idea to wrap it in another callable? Isn't that too wasteful just to shorten its name?
I'd use `std::generate`, but I'm afraid it'll break your "peace of mind", because you'll have to have "tmp" mutable.

Reply:
>Isn't that too wasteful just to shorten its name?

Wasteful in what? one more alias or does it incur performance penalty?
As it is not much difference from using algorithm functions in std lib, I would expect the compiler to inline the inner of lambda and have zero penalty.

Reply:
I picked the original without "using...". I was gonna split the line anyway.

Reply:
Both old code and `std::generate` are quite good scalable algorithms. Sorry, but IMO your suggested code is a special case of absurdity aimed only at making "tmp" variable to be constant.
I agree that having many variables immutable is a good tool to avoid some possible errors. But it seems that for you it is not a "tool", but a "goal in itself". And you are ready to achieve it regardless of the costs. It really bothers me.

Reply:
@Chocobo1, I apologize in advance if my comment above seems offensive to you. It's nothing personal. But once you offer to go by questionable way here, please bother to prove its advantages.

Reply:
>is a special case of absurdity aimed only at making "tmp" variable to be constant.

It is a special case... because only 6 function calls are required and since it is short why not make the tmp const too? If 100 calls  are needed then std::generate will be suitable IMO.
I think this is merely personal preference and you over reacted.

@thalieht 
I don't insist on my suggestion, pick one as you see fit.

Reply:
>I think this is merely personal preference

Following personal preference can be accepted when you produce new code. But making changes of existing things based on subjective preferences is counterproductive. Such an approach can generate a lot of useless changes that simply adjust some part of the code to someone's personal preferences, without bringing any objective benefit. Am I right?